HACKING.md

```diff
@@ -0,0 +1,118 @@
+## Introduction
+LU is build with the [Entity–component–system](https://en.wikipedia.org/wiki/Entity_component_system) which you may know from the Unity Engine. In short that means everything that you see on screen (and more) is a GameObject on which Components can be attached. These components give the object functionality like rendering, physics and so on. 
+
+LU used two different types of servers to communicate. These are: 
+* The authentication server (Uchu.Auth), which handles login and redirection to the world server. 
+* The world server, which handles two things
+  * Character selection (Uchu.Char)
+  * Main game (Uchu.World)
+
+**Note:** A world server in Uchu only handles one world but there can be multiple world servers in one Uchu instance. They can even serve the same map if one world server has reached the maximum amount of players.
+
+The address of the auth server is configured in the `boot.cfg` file in the installation directory of the client. All other servers can have any address because the previous server sends the address to the client.
+
+There were also two other server types that aren't implemented in Uchu. These are
+* The Chat server, which isn't used, since chat messages are handled in the world server
+* The UGC (user generated content) server. This handles stuff like custom icons for modular rockets and cars. The UGC Server address is also set in `boot.cfg`
+
+## The repo structure
+As you probably already noticed the Uchu repository is quite complex. Therefore this paragraph will explain what the different folders contain and what they are for.
+
+There are 3 submodules (marked with a blue text color in GitHub) in the repo.
+* [InfectedRose](https://github.com/Wincent01/InfectedRose) - A library to work with LEGO® Universe files
+* [NexusLogging](https://github.com/TheNexusAvenger/Nexus-Logging) - A logging library for .NET
+* [RakDotNet](https://github.com/UchuServer/RakDotNet) - A library for the RakNet protocol that LU uses
+
+Then there are also the folders that contain the actual Uchu code. These are:
+
+* **Uchu.Api** - It contains code for an api with which the Uchu instance can be controlled. There is a [Cli Tool](https://github.com/UchuServer/cli) that uses this api
+* **Uchu.Auth** - This contains the auth server
```

Comment:
These should probably be clearer because they don't actually contain the auth server but rather the packet handlers and packets that Uchu.Instance use for the auth server. (Similar should go for Uchu.Char and Uchu.World)

Reply:
Tried to make it clearer. Is that better?